src/ethereum/frontier/vm/instructions/memory.py

```diff
@@ -43,8 +48,12 @@ def mstore(evm: Evm) -> None:
     value = pop(evm.stack).to_be_bytes32()
 
     # GAS
-    extend_memory(evm, start_position, U256(len(value)))
-    charge_gas(evm, GAS_VERY_LOW)
+    memory_cost, size_to_extend = calculate_gas_extend_memory(
+        evm.memory, [(start_position, U256(len(value)))]
+    )
+
+    charge_gas(evm, GAS_VERY_LOW + memory_cost)
+    evm.memory += b"\x00" * size_to_extend
```

Comment:
I think `evm.memory += b"\x00" * size_to_extend` belongs in the `# OPERATION` section.